### PR TITLE
Update to latest Next.js features

### PR DESCRIPTION
**Pull request by lfades**
- Update to the latest Next.js versions and removed deprecated code, this includes `componentWillMount`, `props.url` and `prefetch`
- Removed the custom server with Micro
- Moved the dynamic routes for `/plugins/:id` and `/plugins/:id/source` to the new dynamic routes in Next.js 9
- Updated `now.json` and the project now works with `now dev` too
- Fixed issues with the global search, the search was getting stuck sometimes  in a loading state and navigations did not work, also removed the url blink when searching.
- Removed the search page, it's not required because you can search using any page and the layout will take care of it
- The plugins and themes pages are static now, individual plugin pages still use SSR
- The MDX post was moved to a `.mdx` page
- Other small fixes, code reductions and removed unnecessary operations

### CODE REVIEW

**Comment by zeit-team** [comment]
#### 📝Changed routes:


| [**/blog**](https://hyper-site-git-next-latest.zeit.sh/blog) | [**/plugins/\[id\]**](https://hyper-site-git-next-latest.zeit.sh/plugins/[id]) |
|:-:|:-:|
|<a href="https://hyper-site-git-next-latest.zeit.sh/blog">Screenshot of /blog</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/6JygoXXwDfqxBiz_ClarTcOhVqk_">(view full size)</a> |<a href="https://hyper-site-git-next-latest.zeit.sh/plugins/[id]">Screenshot of /plugins/[id]</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/pNJ_c5vzdjzbKJP4hnlnQXHgWIE_">(view full size)</a> |



| [**/plugins/\[id\]/source**](https://hyper-site-git-next-latest.zeit.sh/plugins/[id]/source) | [**/plugins**](https://hyper-site-git-next-latest.zeit.sh/plugins) |
|:-:|:-:|
|<a href="https://hyper-site-git-next-latest.zeit.sh/plugins/[id]/source">Screenshot of /plugins/[id]/source</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/qicyTtjX6kbzp1tQj3sc_TEaNIs_">(view full size)</a> |<a href="https://hyper-site-git-next-latest.zeit.sh/plugins">Screenshot of /plugins</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/JTktzZXtoWwS4lvW_fqf2tKxDog_">(view full size)</a> |



| [**/themes**](https://hyper-site-git-next-latest.zeit.sh/themes) |
|:-:|
|<a href="https://hyper-site-git-next-latest.zeit.sh/themes">Screenshot of /themes</a><br /><sup><a href="https://microlink-cdn.s3.amazonaws.com/s/fpWdxaReHL0CmpjYuR8AnhQrEM0_">(view full size)</a> |





And 4 deleted routes:
- **/blog**
- **/plugin**
- **/plugins**
- **/search**

Commit a960846e3398900fd1da8b8b8bf99bad899fcbec (https://hyper-site-git-next-latest.zeit.sh).

File: next.config.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
-// next.config.js
-const withMDX = require('@zeit/next-mdx')()
+const withMDX = require('@next/mdx')()
+
 module.exports = withMDX({
-  target: 'serverless'
+  target: 'serverless',
+  pageExtensions: ['jsx', 'js', 'ts', 'tsx', 'mdx']
```
Comment (huv1k) [inline next.config.js]:
```suggestion
  pageExtensions: ['js', 'mdx']
```

Comment (lfades) [reply next.config.js]: Makes sense, but any special reason to not keep the other ones to?

Comment (huv1k) [reply next.config.js]: Few `ms` faster build times, because it's not checking all possible extensions.

Comment (lfades) [reply next.config.js]: Good enough

**Pull request merged**